com.unity.ml-agents.extensions/Runtime/Sensors/PoseExtractor.cs

```diff
@@ -62,6 +100,17 @@ protected void SetParentIndices(int[] parentIndices)
         /// <returns></returns>
         protected abstract Pose GetPoseAt(int index);
 
+        /// <summary>
+        /// Return the world space linear velocity of the i'th object.
+        /// </summary>
+        /// <param name="index"></param>
+        /// <returns></returns>
+        protected virtual Vector3 GetLinearVelocityAt(int index)
```

Comment:
should this be abstract?

Reply:
I went with a default implementation in case it was useful for non-physics systems (where you have a hierarchy but not a velocity, say in an animation skeleton). But I can make it abstract instead.